app/gui/src/model/suggestion_database.rs

```diff
@@ -27,6 +28,165 @@ pub use example::Example;
 
 
 
+// ====================
+// === FreeformPath ===
+// ====================
+
+type PathSegment = ImString;
+
+/// A path entered by a User, not guaranteed to correspond to any existing entity. The path is
+/// formed of segments with no guaranteed semantic meaning.
+///
+/// A conversion from a string value is performed by splitting the string into segments separated
+/// by the [`ACCESS`] character.
+#[derive(Clone, Debug, Default)]
+pub struct FreeformPath {
+    segments: Vec<PathSegment>,
+}
+
```

Comment:
As I understand, `FreefromPath` is semantically just "some strings separated by ACCESS". And I see several problems here:
1. Why such generic structure is in suggestion_database module?
2. What is the meaning of converting Entry to "some string separated by access"?
3. What is kept in `freeform_path_to_id_map` field? The "freeform path" term is so generic, that the user has no clue what the map is mapping and why.

So, at least, I would rename it and give it a meaning of being "entry qualified name" - this is how we're using it, isn't it? And, in fact, we can use `impl Iterator<Item=PathSegment>` in each function, omitting some unnecessary copying, so perhaps we could resign from this structure entirely?